resources/qml/UserSettingsPage.qml

```diff
@@ -0,0 +1,218 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+// SPDX-FileCopyrightText: 2022 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import "ui"
+import Qt.labs.platform 1.1 as Platform
+import QtQuick 2.15
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.2
+import QtQuick.Window 2.13
+import im.nheko 1.0
+
+Rectangle {
+    id: userSettingsDialog
+
+    property bool collapsed: width < 800
+    color: Nheko.colors.window
+
+    Shortcut {
+        sequence: StandardKey.Cancel
+        onActivated: userSettingsDialog.close()
+    }
+
+    ScrollView {
+        id: scroll
+
+        palette: Nheko.colors
+        ScrollBar.horizontal.visible: false
+        anchors.fill: parent
+        anchors.margins: Nheko.paddingLarge
+
+        contentWidth: availableWidth
+
+        Timer {
+            id: deadTimer
+            interval: 500
+        }
+
+        Connections {
+            target: scroll.contentItem
+            function onContentYChanged() { deadTimer.restart(); }
+        }
+
+
+        GridLayout {
+            id: grid
+
+            columns: userSettingsDialog.collapsed ? 1 : 2
+            rowSpacing: Nheko.paddingMedium
+            columnSpacing: Nheko.paddingMedium
+
+            anchors.fill: parent
+            anchors.leftMargin: userSettingsDialog.collapsed ? Nheko.paddingLarge : (userSettingsDialog.width-600) * 0.4
+            anchors.rightMargin: userSettingsDialog.collapsed ? Nheko.paddingLarge : (userSettingsDialog.width-600) * 0.4
+
+            Repeater {
+                model: UserSettingsModel
+
+                delegate: Item {
+                    required property var model
+                    id: r
+
+                    Component.onCompleted: {
+                        while (children.length) { 
+                            console.log("Reparenting: " + children[0]);
+                            children[0].parent = grid;
+                        }
+                    }
+
+                    Label {
+                        Layout.alignment: Qt.AlignLeft
+                        Layout.fillWidth: true
+                        color: Nheko.colors.text
+                        text: model.name
+                        //Layout.column: 0
+                        Layout.columnSpan: (model.type == UserSettingsModel.SectionTitle && !userSettingsDialog.collapsed) ? 2 : 1
+                        //Layout.row: model.index
+                        Layout.minimumWidth: implicitWidth
+                        Layout.leftMargin: model.type == UserSettingsModel.SectionTitle ? 0 : Nheko.paddingMedium
+                        Layout.topMargin: model.type == UserSettingsModel.SectionTitle ? Nheko.paddingLarge : 0
+                        font.pointSize: 1.1 * fontInfo.pointSize
+
+                        HoverHandler {
+                            id: hovered
+                            enabled: model.description ?? false
```

Comment:
I had no idea QML supported the null coalescing operator =D

Reply:
It does since 5.15 and the nullchaining in 6.2 or so.